x-pack/plugin/repository-encrypted/src/internalClusterTest/java/org/elasticsearch/repositories/encrypted/EncryptedRepositorySecretIntegTests.java

```diff
@@ -398,14 +396,6 @@ public void testSnapshotIsPartialForMissingPassword() throws Exception {
             equalTo(createSnapshotResponse.getSnapshotInfo().totalShards())
         );
         assertThat(createSnapshotResponse.getSnapshotInfo().successfulShards(), equalTo(0));
-        assertThat(
-            createSnapshotResponse.getSnapshotInfo().userMetadata(),
-            not(hasKey(EncryptedRepository.PASSWORD_HASH_USER_METADATA_KEY))
-        );
-        assertThat(
-            createSnapshotResponse.getSnapshotInfo().userMetadata(),
-            not(hasKey(EncryptedRepository.PASSWORD_SALT_USER_METADATA_KEY))
-        );
```

Comment:
The snapshot metadata is not used anymore to store password hashes.
Instead, the metadata of the password generation, which is stored on the external service, contains the password hash for the password that has been used to encrypt all the DEKs in the generation. The hash is verified by each node before storing a new encrypted DEK in the generation's container.